backends-common/rabbitmq/src/test/java/org/apache/james/backends/rabbitmq/RabbitMQTest.java

```diff
@@ -327,6 +330,137 @@ void rabbitMQShouldSupportTheExclusiveWorkQueueCase() throws Exception {
                 assertThat(fallbackConsumer.getConsumedMessages()).contains(1, 2).doesNotContain(0);
             }
 
+            @Test
+            void rabbitMQShouldDeliverMessageToSingleActiveConsumer() throws Exception {
+                channel1.exchangeDeclare(EXCHANGE_NAME, "direct", DURABLE);
+                channel1.queueDeclare(WORK_QUEUE, DURABLE, !EXCLUSIVE, !AUTO_DELETE, Constants.WITH_SINGLE_ACTIVE_CONSUMER);
+                channel1.queueBind(WORK_QUEUE, EXCHANGE_NAME, ROUTING_KEY);
+
+                IntStream.range(0, 10)
+                    .mapToObj(String::valueOf)
+                    .map(RabbitMQTest.this::asBytes)
+                    .forEach(Throwing.consumer(
+                        bytes -> channel1.basicPublish(EXCHANGE_NAME, ROUTING_KEY, NO_PROPERTIES, bytes)));
```

Comment:
`sneakyThrow`?

Reply:
why do you think it would make a difference?

Reply:
From the documentation:

>  sneakyThrow(): uses the same trick as lombok's @SneakyThrows annotation to rethrow the exception thrown as is, even if checked.

If I understand correctly: the Exception in modified/wrapped without it.